projects/app-lob/src/app/grid-dynamic-chart-data/data-analysis-dock-manager/data-analysis-dock-manager.component.ts

```diff
@@ -163,12 +194,47 @@ export class DataAnalysisDockManagerComponent implements OnInit, AfterViewInit {
         this.paneService.appendPane(splitPane);
         const chartHost = this.getChartHostFromSlot(type);
         chartHost.viewContainerRef.clear();
-        this.chartIntegration.chartFactory
         const chart = this.chartIntegration.chartFactory(type, chartHost.viewContainerRef);
 
         this.dockManager.nativeElement.layout.floatingPanes.push(splitPane);
         this.docLayout = { ...this.dockManager.nativeElement.layout };
         this.selectedCharts[type] = chart;
         this.cdr.detectChanges();
     }
-}
+
+    public createChartCommonLogic() {
+        if (Object.keys(this.selectedCharts).length !== 0) {
+            setTimeout(() => {
+                Object.keys(this.selectedCharts).forEach((c: CHART_TYPE) => {
+                    const chartHost = this.getChartHostFromSlot(c);
+                    if (this.availableCharts.indexOf(c) !== -1) {
+                        if (c !== CHART_TYPE.PIE && typeof this.selectedCharts[c] === 'object') {
+                            this.selectedCharts[c] = this.chartIntegration.chartFactory(c, null, this.selectedCharts[c]);
+                        } else {
+                            chartHost.viewContainerRef.clear();
+                            this.selectedCharts[c] = this.chartIntegration.chartFactory(c, chartHost.viewContainerRef);
```

Comment:
Export the this.selectedCharts[c] into a constant. This is the third usage in few lines.